### PR TITLE
DocC setup

### PR DESCRIPTION
**Pull request by yim-lee**

### CODE REVIEW

**Comment by yim-lee** [comment]
Builds in new Swift versions failing because of `LinuxMain`.

File: Sources/Lifecycle/Lifecycle.swift [hunk 1]
```diff
@@ -34,46 +34,46 @@ public protocol LifecycleTask {
     var logShutdown: Bool { get }
 }
 
-extension LifecycleTask {
-    public var shutdownIfNotStarted: Bool {
+public extension LifecycleTask {
```
Comment (tomerd) [inline Sources/Lifecycle/Lifecycle.swift:37]: intentional?

File: Sources/Lifecycle/Lifecycle.swift [hunk 2]
```diff
@@ -107,8 +107,8 @@ public struct LifecycleHandler {
 
 #if canImport(_Concurrency) && compiler(>=5.5.2)
 @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
-extension LifecycleHandler {
-    public init(_ handler: @escaping () async throws -> Void) {
+public extension LifecycleHandler {
```
Comment (tomerd) [inline Sources/Lifecycle/Lifecycle.swift:110]: intentional?

File: Sources/Lifecycle/Lifecycle.swift [hunk 3]
```diff
@@ -171,8 +171,8 @@ public struct LifecycleStartHandler<State> {
 
 #if canImport(_Concurrency) && compiler(>=5.5.2)
 @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
-extension LifecycleStartHandler {
-    public init(_ handler: @escaping () async throws -> State) {
+public extension LifecycleStartHandler {
```
Comment (tomerd) [inline Sources/Lifecycle/Lifecycle.swift:174]: intentional?

File: Sources/Lifecycle/Lifecycle.swift [hunk 4]
```diff
@@ -233,8 +233,8 @@ public struct LifecycleShutdownHandler<State> {
 
 #if canImport(_Concurrency) && compiler(>=5.5.2)
 @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
-extension LifecycleShutdownHandler {
-    public init(_ handler: @escaping (State) async throws -> Void) {
+public extension LifecycleShutdownHandler {
```
Comment (tomerd) [inline Sources/Lifecycle/Lifecycle.swift:236]: intentional?

File: Sources/Lifecycle/Lifecycle.swift [hunk 5]
```diff
@@ -343,7 +343,7 @@ public struct ServiceLifecycle {
     }
 }
 
-extension ServiceLifecycle {
+public extension ServiceLifecycle {
```
Comment (tomerd) [inline Sources/Lifecycle/Lifecycle.swift:346]: intentional?

File: Sources/Lifecycle/Lifecycle.swift [hunk 6]
```diff
@@ -427,9 +427,9 @@ extension ServiceLifecycle: LifecycleTasksContainer {
     }
 }
 
-extension ServiceLifecycle {
-    /// `ServiceLifecycle` configuration options.
-    public struct Configuration {
+public extension ServiceLifecycle {
```
Comment (tomerd) [inline Sources/Lifecycle/Lifecycle.swift:430]: intentional?

File: Sources/Lifecycle/Lifecycle.swift [hunk 7]
```diff
@@ -745,72 +745,72 @@ extension ComponentLifecycle: LifecycleTasksContainer {
...
+    /// De-register a ``LifecycleTask`` from a ``LifecycleTasksContainer``.
     ///
     /// - parameters:
     ///    - registrationKey: The key returned by a register operation.
     func deregister(_ key: RegistrationKey)
 }
 
-extension LifecycleTasksContainer {
-    /// Register a `LifecycleTask` with a `LifecycleTasksContainer`.
+public extension LifecycleTasksContainer {
```
Comment (tomerd) [inline Sources/Lifecycle/Lifecycle.swift:766]: intentional?

File: Sources/LifecycleNIOCompat/Bridge.swift [hunk 8]
```diff
@@ -34,13 +34,13 @@ extension LifecycleHandler {
     }
 }
 
-extension LifecycleHandler {
+public extension LifecycleHandler {
```
Comment (tomerd) [inline Sources/LifecycleNIOCompat/Bridge.swift:37]: intentional?

File: Sources/LifecycleNIOCompat/Bridge.swift [hunk 9]
```diff
@@ -49,12 +49,12 @@ extension LifecycleHandler {
     }
 }
 
-extension LifecycleStartHandler {
+public extension LifecycleStartHandler {
```
Comment (tomerd) [inline Sources/LifecycleNIOCompat/Bridge.swift:52]: intentional?

File: Sources/LifecycleNIOCompat/Bridge.swift [hunk 10]
```diff
@@ -63,12 +63,12 @@ extension LifecycleStartHandler {
     }
 }
 
-extension LifecycleShutdownHandler {
+public extension LifecycleShutdownHandler {
```
Comment (tomerd) [inline Sources/LifecycleNIOCompat/Bridge.swift:66]: intentional?

File: Sources/LifecycleNIOCompat/Bridge.swift [hunk 11]
```diff
@@ -82,13 +82,13 @@ extension LifecycleShutdownHandler {
     }
 }
 
-extension ComponentLifecycle {
+public extension ComponentLifecycle {
```
Comment (tomerd) [inline Sources/LifecycleNIOCompat/Bridge.swift:85]: intentional?

File: Sources/LifecycleNIOCompat/Bridge.swift [hunk 12]
```diff
@@ -15,12 +15,12 @@
 import Lifecycle
 import NIO
 
-extension LifecycleHandler {
+public extension LifecycleHandler {
```
Comment (tomerd) [inline Sources/LifecycleNIOCompat/Bridge.swift:18]: intentional?

File: scripts/soundness.sh [hunk 13]
```diff
@@ -73,7 +73,7 @@ for language in swift-or-c bash dtrace; do
   matching_files=( -name '*' )
   case "$language" in
       swift-or-c)
-        exceptions=( -name Package.swift )
+        exceptions=( -name Package.swift -o -name Package@swift-5.6.swift )
```
Comment (tomerd) [inline scripts/soundness.sh:76]: `Package@swift-5.6.swift` -> `Package@*.swift`

File: Sources/Lifecycle/Documentation.docc/index.md [hunk 14]
```diff
@@ -0,0 +1,311 @@
+# ``Lifecycle``
```
Comment (tomerd) [inline Sources/Lifecycle/Documentation.docc/index.md:1]: https://github.com/apple/swift-statsd-client/pull/36/files#r934924113

**Review by tomerd — APPROVED** [review]

File: Package@swift-5.6.swift [hunk 15]
```diff
@@ -0,0 +1,34 @@
+// swift-tools-version:5.6
```
Comment (tomerd) [inline Package@swift-5.6.swift:1]: until we hear more back from @fabianfett, let go with the "old files style", i.e make Package.swift required 5.6 and add files for each one of the old version we support.

**Pull request merged**